extensions/ql-vscode/src/common/app.ts

```diff
@@ -23,10 +23,15 @@ export interface App {
   readonly onDidChangeWorkspaceFolders: Event<WorkspaceFoldersChangeEvent>;
   readonly credentials: Credentials;
   readonly commands: AppCommandManager;
+  readonly environment: EnvironmentContext;
```

Comment:
I'm wondering how we should decide whether something belongs in `environment` or in the `App` itself. Right now it reflects `vscode.env`, but it doesn't seem like there are other properties which we would need in the extension. Would there be any advantages to adding it as `App.language`? That does mean the complete `App` object needs to be passed around, but this seems to already be happening in a lot of places anyway.

I'm definitely in favour of splitting up the `App` object, I'm just not sure when we should split it up, and when we should add it to the `app` directly (like `workspaceFolders`).

Reply:
Its a good question and I think it'll generally be difficult to answer. To me it makes sense to try and group things somewhat so that we don't pass the whole App object when we don't need it.  

For `environment` I can imagine other things like timezone information being added to it.

Reply:
I've opened https://github.com/github/vscode-codeql/pull/2481 to show what the `Localizer` option would look like.

Reply:
Thanks for trying it out! I think I prefer this version for now because it keeps all the VSCode and testing specific things in the single `App` interface. I'm fine to go ahead with this PR, this is not a block consideration from my view.

Reply:
FWIW I also prefer the current implementation but it's great to see the other one in action! Thanks Robert!